### PR TITLE
Fix several small issues with rule docs

### PR DESCRIPTION
**Pull request by barrywhart**
### Brief summary of the change made

This PR fixes a large number of issues with rule documentation, mainly inconsistent or incorrect quoting of code samples, but also some wording, etc.

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

**Review by tunetheweb — APPROVED** [review]

File: src/sqlfluff/core/rules/config_info.py [hunk 1]
```diff
@@ -65,7 +65,7 @@
         "definition": (
             "The capitalisation policy to enforce, extended with PascalCase. "
             "This is separate from capitalisation_policy as it should not be "
-            "applied to keywords."
+            "applied to keywords"
```
Comment (jpy-git) [inline src/sqlfluff/core/rules/config_info.py:68]: @barrywhart question about this one. I think we should keep the full stops here and remove it from whatever template it's getting subbed into.

Reason is that some config descriptions end with question marks (e.g. `multiline_newline`) so we have a weird `?.` in the docs:
image

File: src/sqlfluff/core/rules/doc_decorators.py [hunk 2]
```diff
@@ -51,7 +51,7 @@ def document_configuration(cls, ruleset="std"):
                 keyword, info_dict["definition"]
             )
             if "validation" in info_dict:
-                config_doc += " Must be one of {}.".format(info_dict["validation"])
+                config_doc += " Must be one of ``{}``.".format(info_dict["validation"])
```
Comment (jpy-git) [inline src/sqlfluff/core/rules/doc_decorators.py:54]: for my reference, are double backticks like the single backtick code blocks in markdown or something different?

File: src/sqlfluff/rules/L028.py [hunk 3]
```diff
@@ -24,7 +24,7 @@ class Rule_L028(Rule_L025):
         FROM foo
 
     | **Best practice**
-    |  Remove all the reference or reference all the fields.
+    |  Remove or add references to all fields.
```
Comment (jpy-git) [inline src/sqlfluff/rules/L028.py:27]:
```suggestion
    |  Add or remove references to all fields.
```
NIT: reads weird the other way round

File: src/sqlfluff/rules/L043.py [hunk 4]
```diff
@@ -39,7 +39,8 @@ class Rule_L043(BaseRule):
             end as is_fab
         from fancy_table
 
-        -- We can also simplify CASE statements that aim to fill NULL values.
+        -- This rule can also simplify ``CASE`` statements that aim to fill
+        -- NULL values.
```
Comment (jpy-git) [inline src/sqlfluff/rules/L043.py:43]:
```suggestion
        -- This rule can also simplify ``CASE`` statements
        -- that aim to fill NULL values.
```
looks a bit odd having such a small portion of the sentence on it's own line.

File: src/sqlfluff/core/rules/doc_decorators.py (hunk 2 above)
Comment (tunetheweb) [reply src/sqlfluff/core/rules/doc_decorators.py:54]: Yup. Had to look this one up myself to be sure:

https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html

Comment (barrywhart) [reply src/sqlfluff/core/rules/doc_decorators.py:54]: Just learned this myself today!

File: src/sqlfluff/core/rules/config_info.py (hunk 1 above)
Comment (barrywhart) [reply src/sqlfluff/core/rules/config_info.py:68]: Will do. I found the place and will update it.

File: src/sqlfluff/rules/L043.py (hunk 4 above)
Comment (barrywhart) [reply src/sqlfluff/rules/L043.py]: Changed.

File: src/sqlfluff/rules/L028.py (hunk 3 above)
Comment (barrywhart) [reply src/sqlfluff/rules/L028.py]: Updated.

**Comment by barrywhart** [comment]
Thanks for the review, guys! I've addressed all the feedback. Merging now.

**Comment by jpy-git** [comment]
@barrywhart Whilst we're on the topic of fixing rule docs:

Is it just me or does this line `Register all the rule classes with their corresponding rulesets (just std currently).` before the rule docs read weird?
image

Doesn't make much sense to me let alone a user. Probably should just say something like "Documentation and examples of individual rules" or not even be there?

**Comment by barrywhart** [comment]
@jpy-git: Yes, it's weird. This is sort of an auto-doc issue -- it comes from this section of `docs/rules.rst`:

```
Specific Rules
--------------

.. automodule:: sqlfluff.core.rules
   :members:
   :member-order: alphabetical
```

The `automodule` directive is picking up the module docstring in `src/sqlfluff/core/rules/__init__.py`. It makes sense as a docstring, but not for displaying in user-facing docs. I already merged this PR -- would you like to create a PR with that change? It sounds good to me.

**Comment by jpy-git** [comment]
Makes sense! Will pick this up tomorrow 😊

**Pull request merged**